tensorpipe/channel/cuda_xth/channel_impl.h

```diff
@@ -62,25 +65,27 @@ struct RecvOperation {
   bool doneReadingDescriptor{false};
 
   // Arguments at creation
+  bool isCudaBuffer;
   void* const ptr;
   const size_t length;
-  const int deviceIdx;
-  const cudaStream_t stream;
+  int deviceIdx;
+  cudaStream_t stream;
   TRecvCallback callback;
 
   // Other data
-  cudaEvent_t startEvent;
+  bool isSrcCudaBuffer;
+  cudaEvent_t startEv;
   const void* srcPtr;
   int srcDeviceIdx;
   cudaStream_t srcStream;
 
+  RecvOperation(CpuBuffer buffer, size_t length, TRecvCallback callback);
+
   RecvOperation(
       int deviceIdx,
       CudaBuffer buffer,
```

Comment:
It's an odd asymmetry that in RecvOp we take a CudaBuffer whereas in SendOp we take void* + cudaStream_t

Reply:
![Uploading gameshare.png…]()
#6